lib/routes/pos/home/pos_invoice.dart

```diff
@@ -390,56 +434,87 @@ class POSInvoiceState extends State<POSInvoice> {
   onAddition() {
     setState(() {
       _totalAmount += _currentAmount;
-      _chargeAmountController.text = _currency.format((Int64(_totalAmount)),
-          fixedDecimals: true, includeSymbol: true);
-      _currentAmount = 0;
-      _amountController.text = _currency.format((Int64(_currentAmount)),
-          fixedDecimals: true, includeSymbol: false);
+      _clearCurrentAmounts();
+      _updateAmountControllers();
     });
   }
 
   onNumButtonPressed(String numberText) {
     setState(() {
-      _currentAmount = (_currentAmount * 10) + int.parse(numberText);
-      _amountController.text = _currency.format((Int64(_currentAmount)),
-          fixedDecimals: true, includeSymbol: false);
-      _chargeAmountController.text = _currency.format(
-          (Int64(_totalAmount + _currentAmount)),
-          fixedDecimals: true,
-          includeSymbol: true);
+      if (!_usingFiat) {
+        _currentAmount = (_currentAmount * 10) + int.parse(numberText);
+      } else {
+        _currentFiatAmount = _currentFiatAmount * 10 +
+            int.parse(numberText) /
+                pow(10, _fiatCurrencyConversion.currencyData.fractionSize);
+        _currentAmount =
+            _fiatCurrencyConversion.fiatToSat(_currentFiatAmount).toInt();
+      }
+      _updateAmountControllers();
     });
   }
 
   changeCurrency(value) {
     setState(() {
-      _currency = Currency.fromSymbol(value);
-      _userProfileBloc.currencySink
-          .add(Currency.currencies[Currency.currencies.indexOf(_currency)]);
+      bool usingFiat = _fiatCurrencyConversion != null ? true : false;
```

Comment:
Could this be null? Because we don't check it in other places at this file.

Reply:
It can be null if getExchangeRates fails.
[Removed that logic](https://github.com/breez/breezmobile/pull/268/commits/4cfbca6e14e6a9038239d47a42bf0721ef0049ac)